### PR TITLE
ID-3359 [FIX] Ensure Matrix field's default value display correctly when "User's profile" type is selected

### PR DESCRIPTION
**Pull request by NikitaJadhavXoriant**
**Product areas affected**

Studio -> Form -> Form Fields -> Matrix field

**What does this PR do?**

Implemented code, so Matrix field's default value display correctly when "User's profile" type is selected

**JIRA ticket**

[click here](https://weboo.atlassian.net/browse/ID-3359)

**Result**

https://github.com/Fliplet/fliplet-widget-form-builder/assets/108272606/298dc34b-8f90-4238-8560-7f1d4133951e

**Checklist**

None

**Testing instructions**

None

**Deployment instructions**

None

**Author concerns**

None

### CODE REVIEW

File: js/libs/form.js [hunk 1]
```diff
@@ -1017,12 +1022,26 @@ Fliplet().then(function() {
...
                   if (!_.isEmpty(value)) {
-                    _.forEach(value, function(col, row) {
-                      if (!row || !col) {
-                        return '';
-                      }
+                    _.forEach(field.rowOptions, function(rowOpt) {
+                      var rowFlag = false;
+                      var val = rowOpt.id ? rowOpt.id : rowOpt.label;
+
+                      _.some(value, function(col, row) {
```
Comment (tonytlwu) [inline js/libs/form.js:1029]:
```suggestion
                      var rowFound = _.some(value, function(col, row) {
```

File: js/libs/form.js [hunk 2]
```diff
@@ -1017,12 +1022,26 @@ Fliplet().then(function() {
...
+                        if (val === row) {
+                          appendField(`${field.name} [${val}]`, col);
+                          rowFlag = true;
+
+                          return true;
+                        }
+                      });
 
-                      appendField(`${field.name} [${row}]`, col);
+                      if (!rowFlag) {
```
Comment (tonytlwu) [inline js/libs/form.js:1042]:
```suggestion
                      if (!rowFound) {
```

File: js/libs/form.js [hunk 3]
```diff
@@ -1017,12 +1022,26 @@ Fliplet().then(function() {
                   }
                 } else if (type === 'flMatrix') {
                   if (!_.isEmpty(value)) {
-                    _.forEach(value, function(col, row) {
-                      if (!row || !col) {
-                        return '';
-                      }
+                    _.forEach(field.rowOptions, function(rowOpt) {
+                      var rowFlag = false;
+                      var val = rowOpt.id ? rowOpt.id : rowOpt.label;
```
Comment (tonytlwu) [inline js/libs/form.js:1027]:
```suggestion
                      var val = rowOpt.id || rowOpt.label;
```

**Review by tonytlwu — CHANGES_REQUESTED** [review]

**Review by tonytlwu — CHANGES_REQUESTED** [review]

File: js/libs/form.js [hunk 4]
```diff
@@ -254,7 +254,7 @@ Fliplet().then(function() {
         fields.forEach(function(field) {
           if (entry && entry.data && field.populateOnUpdate !== false) {
             var fieldKey = isResetAction
-              ? field.defaultValueKey || field.name
+              ? field.defaultValueKey
```
[the remark this answers is not in the document]
Comment (NikitaJadhavXoriant) [reply js/libs/form.js:257]: Hi @tonytlwu, yes I have verified this change, as per this [flow](https://docs.google.com/presentation/d/1ZcDdKZthUujPP_VuBXX9J5fj2HWTdrsBAdarndBgmnM/edit#slide=id.g1fb36f71a8b_0_1), we need to reset to defaultValueKey and if defaultValueKey is undefined we need to set the value as blank and not to field.name that is why I removed that. Yes, I will ask Kathi for regression testing.

File: js/libs/form.js [hunk 5]
```diff
@@ -291,14 +291,18 @@ Fliplet().then(function() {
 
                 _.forEach(field.rowOptions, function(row) {
                   var val = row.id ? row.id : row.label;
-                  var matrixKey = entry.data[`${fieldKey} [${val}]`] ? entry.data[`${fieldKey} [${val}]`] : entry.data[`${fieldKey}`];
+                  var matrixKey = entry.data[`${fieldKey} [${val}]`] || entry.data[`${fieldKey}`];
 
                   if (isResetAction) {
-                    if ((!field.defaultValueKey && matrixKey) || (field.defaultValueKey.indexOf(val) !== -1 && matrixKey)) {
+                    if ((!field.defaultValueKey && matrixKey) || (field.defaultValueKey.indexOf(val) !== -1 && matrixKey)
```
[the remark this answers is not in the document]
Comment (NikitaJadhavXoriant) [reply js/libs/form.js:297]: Done

**Review by tonytlwu — APPROVED** [review]
@inna-bieshulia This is ready for testing

Skip compile: 
Includes translation: 
Initiative: 
Impact: High
Affected: All field types that support default value settings
Verify: All types of default value configurations are working on all field types and the resetting behaviours are correct

**Pull request merged**